protocol/vm/crypto_test.go

```diff
@@ -84,7 +82,7 @@ func TestCheckSig(t *testing.T) {
 			if err != nil {
 				t.Errorf("case %d: expected ok result, got error %s", i, err)
 			}
-		} else if !vm.FalseResult() {
```

Comment:
I think this PR also fixes a couple of test bugs, such as this line. Pretty sure this should have been saving a gotVM variable on line 78 and using that here as `gotVM.FalseResult`.

The new code I believe is correct, because the existing `vm` variable is updated.